### DISCUSSION TITLE
Streaming to iPads

### DISCUSSION THREAD

**Discussion by gregfr** [OvenMediaLabs/OvenMediaEngine#1171] [Help]
Greetings

I'm trying to stream something with OME so it can be watched on an iPad... I've tried many many possible configurations but I can't figure out how have ABR working correctly.

My last test was to mimic a test stream form HLS.js which seems to be working:

_(I removed the file pathes which I guess are irrelevant)_

```
#EXTM3U  
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=2149280,CODECS="mp4a.40.2,avc1.64001f",RESOLUTION=1280x720,NAME="720"  
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=246440,CODECS="mp4a.40.5,avc1.42000d",RESOLUTION=320x184,NAME="240"  
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=460560,CODECS="mp4a.40.5,avc1.420016",RESOLUTION=512x288,NAME="380"  
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=836280,CODECS="mp4a.40.2,avc1.64001f",RESOLUTION=848x480,NAME="480"  
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=6221600,CODECS="mp4a.40.2,avc1.640028",RESOLUTION=1920x1080,NAME="1080"  
```

Here is the result of my configuration for OME:

```
#EXTM3U  
#EXT-X-VERSION:7  
#EXT-X-INDEPENDENT-SEGMENTS  
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="aac_128",NAME="Audio_1",DEFAULT=YES,AUTOSELECT=YES,CHANNELS="2",URI="/bbb/test/chunklist_5_audio_Iz6NCBXZ_llhls.m3u8"  
  
#EXT-X-STREAM-INF:BANDWIDTH=2149280,RESOLUTION=1280x720,FRAME-RATE=25.000,CODECS="avc1.640c1f,mp4a.40.2",AUDIO="aac_128"  
#EXT-X-STREAM-INF:BANDWIDTH=246440,RESOLUTION=320x184,FRAME-RATE=25.000,CODECS="avc1.42c00c,mp4a.40.2",AUDIO="aac_128"  
#EXT-X-STREAM-INF:BANDWIDTH=460560,RESOLUTION=512x288,FRAME-RATE=25.000,CODECS="avc1.42c015,mp4a.40.2",AUDIO="aac_128"  
#EXT-X-STREAM-INF:BANDWIDTH=836280,RESOLUTION=848x480,FRAME-RATE=25.000,CODECS="avc1.640c1e,mp4a.40.2",AUDIO="aac_128"  
#EXT-X-STREAM-INF:BANDWIDTH=6221600,RESOLUTION=1920x1080,FRAME-RATE=25.000,CODECS="avc1.640c28,mp4a.40.2",AUDIO="aac_128"
```

Same bitrates, same order, but the iPad is most of the times stuck in a lower rendition... I say most of the times because sometimes, some iPads correctly switch to HD...
BTW it's also happening on MacOS Safari...

What am I doing wrong? what renditions do you use for iPads?

Thanks in advance

Regards

**Comment by getroot** [comment]
It is the player's role to choose a Rendition. They choose a rendition based on the Bandwidth and Resolution provided by the master playlist.

It's easy to overlook, but resolution is very important to a player's choice. In fact, they don't bother choosing high-resolution renditions on small players. So switching to full screen often switches to a higher rendition. But the most important thing is the bandwidth. If the network is slow, players will choose a lower rendition regardless of the screen size.

It's a good idea to check that the player isn't picking up a high rendition due to the playback screen size and network speed.

**Comment by gregfr** [comment]
Thanks for your answer. That is exactly my point: considering the playback conditions are the same (same iPad 2 048 x 1 536, same  very good wifi, same 1G fiber network, and now very similar playlist parameters), why does the same iPad stays on the lower rendition when the stream is sent from OME, but switches correctly when sent from something else? I know the answer is not obvious (especially considering that it's Apple's player code), but I've been testing for months now, with no success...

Do you have a rendition map that works for you on iPad?

Thanks in advance

Regards

**Reply by getroot** [reply]
> why does the same iPad stays on the lower rendition when the stream is sent from OME, but switches correctly when sent from something else?

What server is the "something else" you mentioned? And is it LLHLS? Is it installed on the same server as OME?

**Reply by gregfr** [reply]
"something else" is my own HLS server (based on ffmpeg, running on the same machine as OME but not at the same time), but I also tried Vimeo.
I know they're not LLHLS :) my goal is precisely to switch from "anything else" to OME, so I'm looking for a way to configure OME and throw away everything else :)
